### PR TITLE
[WIP - DO NOT MERGE] Adds check_remote and tests for the same

### PR DESCRIPTION
**Pull request by tkell**
Don't merge, we might just use Requests!

(But this is the first step of #34, and it is bad because making a HEAD request is totally different in py2 and py3)

### CODE REVIEW

File: mirdata/utils.py [hunk 1]
```diff
@@ -167,6 +176,36 @@ def update_to(self, b=1, bsize=1, tsize=None):
...
+        Named tuple containing remote dataset meta information: url, filename
+        and checksum.
+
+    Returns
+    -------
+    file_exists: bool
+        True if file exists.
+        False otherwise.
+    """
+    req = Request(remote.url, method='HEAD')
```
Comment (drubinstein) [inline mirdata/utils.py]: can make this a bit neater I think
```
try:
  res = urlopen(req)
  return res.status >= 200 and res.status < 300
except HTTPError:
  pass
return False

File: tests/test_utils.py [hunk 2]
```diff
@@ -60,6 +60,29 @@ def test_get_save_path(mocker, tmpdir):
...
+    httpserver.serve_content('File not found!', 404)
+
+    TEST_META = utils.RemoteFileMetadata(
+        filename='remote.wav',
+        url=httpserver.url,
+        checksum=('3f77d0d69dc41b3696f074ad6bf2852f')
+    )
+
+    file_exists = utils.check_remote(TEST_META)
+    assert file_exists == False
```
Comment (drubinstein) [inline tests/test_utils.py]: `assert not file_exists`

File: tests/test_utils.py [hunk 3]
```diff
@@ -60,6 +60,29 @@ def test_get_save_path(mocker, tmpdir):
...
+    httpserver.serve_content(open('tests/resources/remote.wav').read())
+
+    TEST_META = utils.RemoteFileMetadata(
+        filename='remote.wav',
+        url=httpserver.url,
+        checksum=('3f77d0d69dc41b3696f074ad6bf2852f')
+    )
+
+    file_exists = utils.check_remote(TEST_META)
+    assert file_exists == True
```
Comment (drubinstein) [inline tests/test_utils.py]: `assert file_exists`

File: mirdata/utils.py [hunk 4]
```diff
@@ -17,6 +17,15 @@
     from urllib.request import urlretrieve  # py3
 except ImportError:
     from urllib import urlretrieve  # py2
+try:
```
Comment (drubinstein) [inline mirdata/utils.py]: You can probably merge these (and all the try..excepts) from the previous blocks. Or maybe we should just do a check on the system major version and do
if sys.version.major >= 3
   do py3 imports
else
 do py2 imports

**Comment by tkell** [comment]
The Py2 / Py3 stuff here is a total disaster, and getting worse, yep.  Does `requests` have a better cross-version API?  (cc @rabitt, are you opposed to `requests`?)

**Comment by tkell** [comment]
Or `from six.moves import urllib`, maybe, again cc @rabitt?

**Comment by drubinstein** [comment]
You can use requests. It's py2/3 compat https://pypi.org/project/requests/ look in the lower left hand corner.

**Comment by tkell** [comment]
Closing to do things with Requests, then do the reduced version of this;  see #34 for discussion!

**Pull request closed without merging**